packages/openapi-generator/src/sdk-metadata/code-sample-util.ts

```diff
@@ -0,0 +1,167 @@
+): OpenApiOperation | undefined {
+  let closestOperation: OpenApiOperation | undefined;
+  let minDistance = distanceThreshold;
+
+  operations.forEach(operation => {
+    const distance = getLevensteinDistance(apiName, operation.operationId);
+    if (distance <= minDistance) {
+      minDistance = distance;
+      closestOperation = operation;
+    }
+  });
+  return closestOperation;
+}
+
+export function getGetAllOperation(
+  operations: OpenApiOperation[]
+): OpenApiOperation | undefined {
+  return operations.find(
+    operation =>
+      operation.operationId.toLowerCase() === 'getall' &&
+      operation.pathParameters.length === 0
+  );
+}
+
+export function getGetOperation(
+  operations: OpenApiOperation[]
+): OpenApiOperation | undefined {
+  const getOperations = operations.filter(
+    operation => operation.method === 'get'
+  );
+
+  // check if get operation without parameters exists
+  // else return operation with min parameters
+  if (getOperations.length > 0) {
+    const getOperationsWithoutParams = getOperations.filter(
+      getOperation =>
+        getOperation.pathParameters.length === 0 &&
+        getOperation.queryParameters.length === 0
+    );
+    if (getOperationsWithoutParams.length > 0) {
+      return getOperationsWithoutParams[0];
+    }
+    return getOperations[0];
+  }
+  return undefined;
+}
+
+export function getAnyOperationWithoutParams(
+  operations: OpenApiOperation[]
+): OpenApiOperation | undefined {
+  return operations.find(
+    operation =>
+      operation.pathParameters.length === 0 &&
+      operation.queryParameters.length === 0
+  );
+}
+
+export function getOperationParamCode(operation: OpenApiOperation): string {
+  if (
+    operation.pathParameters?.length === 0 &&
+    operation.queryParameters?.filter(param => param.required).length === 0 &&
+    operation.requestBody === undefined
+  ) {
+    return '';
+  }
+
+  const paramSignature: string[] = [];
+  if (operation.pathParameters?.length > 0) {
+    paramSignature.push(
+      operation.pathParameters.map(pathParam => pathParam.name).join(', ')
+    );
+  }
+
+  if (operation.requestBody) {
+    let reqBody = 'undefined';
+    if (operation.requestBody.required) {
+      reqBody = 'myRequestBody';
+    }
+    paramSignature.push(reqBody);
+  }
+
+  if (operation.queryParameters?.length > 0) {
+    const queryParams = operation.queryParameters
+      .filter(queryParam => queryParam.required)
+      .map(
+        queryParam => `${queryParam.name}: 'my${pascalCase(queryParam.name)}'`
+      );
+
+    if (queryParams) {
+      paramSignature.push(`{ ${queryParams.join(', ')} }`);
+    }
+  }
+
+  return codeBlock`${paramSignature.join(', ')}`;
+}
+
+/**
+ * Calculate levenshtein distance of the two strings.
+ * @param stringA - The first string.
+ * @param stringB - The second string.
+ * @returns The levenshtein distance (0 and above).
+ * @hidden
+ */
+function getLevensteinDistance(stringA: string, stringB: string): number {
```

Comment:
There is a generator-common package which also contains some metadata parts. You could put this method in there. Potentially also a few of the more general method from above after you changed the signature.